### PR TITLE
[URP] DepthCopy texture as a color

### PR DESCRIPTION
**Pull request by jonuuukas**
---
### Purpose of this PR
Why is this PR needed, what hard problem is it solving/fixing?
---
This PR changes the behaviour of URP's DepthCopy pass to output R32_SFloat texture instead of a native depth texture.
Why? Because Native RenderPasses support only a single depth buffer and this copy buffer usually is the main offender on why RenderPasses get split.
This should also enable some features(f.e. SSAO, Decals and etc.) to be able to utilise RenderPass API as well, as a lot of them rely on additional depth texture. This would also mean that the depth could technically be framebuffer-fetched using this newly changed texture

### Testing status
Yamato is as green as it can get (it seems, not sure), but QA pass definitely should be done, as this affects the whole URP. Manually checked all of the URP test suites and ran locally on mac editor.
---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

File: com.unity.render-pipelines.universal/Runtime/Passes/CopyDepthPass.cs [hunk 1]
```diff
@@ -43,14 +43,15 @@ public void Setup(RenderTargetHandle source, RenderTargetHandle destination)
         public override void OnCameraSetup(CommandBuffer cmd, ref RenderingData renderingData)
         {
             var descriptor = renderingData.cameraData.cameraTargetDescriptor;
-            descriptor.colorFormat = RenderTextureFormat.Depth;
-            descriptor.depthBufferBits = 32; //TODO: do we really need this. double check;
+            // descriptor.colorFormat = RenderTextureFormat.Depth;
+            descriptor.graphicsFormat = GraphicsFormat.R32_SFloat;
+            descriptor.depthBufferBits = 0; //TODO: do we really need this. double check;
```
Comment (manuele-bonanno) [inline com.unity.render-pipelines.universal/Runtime/Passes/CopyDepthPass.cs:48]: this looks weird, not sure is needed

File: com.unity.render-pipelines.universal/Runtime/Passes/CopyDepthPass.cs [hunk 2]
```diff
@@ -43,14 +43,15 @@ public void Setup(RenderTargetHandle source, RenderTargetHandle destination)
         public override void OnCameraSetup(CommandBuffer cmd, ref RenderingData renderingData)
         {
             var descriptor = renderingData.cameraData.cameraTargetDescriptor;
-            descriptor.colorFormat = RenderTextureFormat.Depth;
-            descriptor.depthBufferBits = 32; //TODO: do we really need this. double check;
+            // descriptor.colorFormat = RenderTextureFormat.Depth;
```
Comment (manuele-bonanno) [inline com.unity.render-pipelines.universal/Runtime/Passes/CopyDepthPass.cs:46]: the commented out line can be removed

File: com.unity.render-pipelines.universal/Runtime/Passes/DepthOnlyPass.cs [hunk 3]
```diff
@@ -31,6 +31,7 @@ public DepthOnlyPass(RenderPassEvent evt, RenderQueueRange renderQueueRange, Lay
             base.profilingSampler = new ProfilingSampler(nameof(DepthOnlyPass));
             m_FilteringSettings = new FilteringSettings(renderQueueRange, layerMask);
             renderPassEvent = evt;
+            useNativeRenderPass = false;
```
Comment (manuele-bonanno) [inline com.unity.render-pipelines.universal/Runtime/Passes/DepthOnlyPass.cs:34]: why is this pass not RP compatible anymore?

**Review by manuele-bonanno — APPROVED** [review]
looks good! Some minor comments

File: com.unity.render-pipelines.universal/Runtime/Passes/FinalBlitPass.cs [hunk 4]
```diff
@@ -108,7 +108,7 @@ public override void Execute(ScriptableRenderContext context, ref RenderingData
                         cameraTarget,
                         RenderBufferLoadAction.Load,
                         RenderBufferStoreAction.Store,
-                        ClearFlag.None,
+                        ClearFlag.Depth,
```
Comment (phi-lira) [inline com.unity.render-pipelines.universal/Runtime/Passes/FinalBlitPass.cs:111]: I don't follow why we are changing this. Could this maybe introduce regression for users that might read depend on depth after final blit pass?

**Review by phi-lira — APPROVED** [review]

**Comment by phi-lira** [comment]
Adding @PaulDemeulenaere and @thomas-zeng for area review.

**Comment by manuele-bonanno** [comment]
please also update the changelog

**Review by thomas-zeng — APPROVED** [review]
Approve for the XR related changes. Please address other reviewer's feedback :)

**Review by PaulDemeulenaere — APPROVED** [review]
Yamato is green on VFX: https://yamato.cds.internal.unity3d.com/jobs/902-Graphics/tree/universal%252Fdepth-copy-as-color/.yamato%252Fall-vfx.yml%2523VFX_trunk/9097612/job

**Pull request merged**